src/merkle/partial_mt/mod.rs

```diff
@@ -327,3 +350,56 @@ impl PartialMerkleTree {
         Ok(())
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for PartialMerkleTree {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        // max_depth
+        target.write_u8(self.max_depth);
+
+        // nodes
+        target.write_u64(self.nodes.len() as u64);
+        for (index, hash) in self.nodes.iter() {
+            index.write_into(target);
+            hash.write_into(target);
+        }
+
+        // leaves
+        target.write_u64(self.leaves.len() as u64);
+        for elem in self.leaves.iter() {
+            elem.write_into(target);
+        }
+    }
```

Comment:
Do we need to serialize all nodes? It seems like we should be able to just serialize leaf nodes and this is enough to reconstruct the whole set of nodes.